### PR TITLE
Authentications PR.

### LINKED ISSUE #9: Log-in and Sign-up Activities
Integrate Firebase. 
Implement Sign Up Activity
Implement Log in Activity 
Implement Google Sign in 
Implement Facebook Sign in

### PR DESCRIPTION
**Pull request by Sarthak2601**
Fixes #9 
*Connected to Firebase 
*Login via Google 
*Login via Facebook 
*Login via Email/Password 
*SignUp via Email/Password

Screenshot_2019-12-09-22-59-26-448_io github njackwinterofcode translatex

### CODE REVIEW

**Comment by somenath1435** [comment]
@Sarthak2601 remove unnecessary commits. Also change PR title to a descriptive one. Also, for any UI change, always attach screenshots in PR

File: app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,7 +1,11 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
     package="io.github.njackwinterofcode.translatex">
 
+    <uses-permission android:name="android.permission.INTERNET"/>
+
```
Comment (ShivangiSingh17) [inline app/src/main/AndroidManifest.xml]: Remove unnecessary spacing given

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -22,6 +26,28 @@
...
+            android:name="com.facebook.CustomTabActivity"
+            android:exported="true">
+            <intent-filter>
+                <action android:name="android.intent.action.VIEW" />
+                <category android:name="android.intent.category.DEFAULT" />
+                <category android:name="android.intent.category.BROWSABLE" />
+                <data android:scheme="@string/fb_login_protocol_scheme" />
+            </intent-filter>
+        </activity>
+
```
Comment (ShivangiSingh17) [inline app/src/main/AndroidManifest.xml]: As commented above

**Review by ShivangiSingh17 — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (Sarthak2601) [reply app/src/main/AndroidManifest.xml]: @ShivangiSingh17 Done.

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (Sarthak2601) [reply app/src/main/AndroidManifest.xml]: @ShivangiSingh17 Done.

**Pull request closed without merging**